content/utilities/map-export/plugin.js

```diff
@@ -110,7 +112,40 @@ class Plugin {
       this.status.style.color = 'red';
     }
   }
-
+  
+  onDownload = async () => {
+    let chunks = ui.getExploredChunks();
+    let chunksAsArray = Array.from(chunks);
+    if (this.beginCoords && this.endCoords) {
+      let begin = {
+        x: Math.min(this.beginCoords.x, this.endCoords.x),
+        y: Math.max(this.beginCoords.y, this.endCoords.y),
+      };
+      let end = {
+        x: Math.max(this.beginCoords.x, this.endCoords.x),
+        y: Math.min(this.beginCoords.y, this.endCoords.y),
+      };
+      chunksAsArray = chunksAsArray.filter((chunk) => {
+        return this.intersectsXY(chunk, begin, end);
+      });
+    }
+    try {
+      let map = JSON.stringify(chunksAsArray);
+      var blob = new Blob([map], { type: 'text/plain' }),
+          anchor = document.createElement('a');
+      anchor.download = "hello.txt";
+      anchor.href = (window.webkitURL || window.URL).createObjectURL(blob);
+      anchor.dataset.downloadurl = ['text/plain', anchor.download, anchor.href].join(':');
+      anchor.click();
```

Comment:
I wonder if this should be split into a button saying "generate map file" and then it inserts the link which says "download map"

Reply:
I separated the repetitive code into a generateMap() function, but didn't add a button. the generation seems to be quick.